### ISSUE TITLE
hansfriese - Can not change the fee of the Redeemer

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2022-10-illuminate-judging#132]
hansfriese

medium

# Can not change the fee of the Redeemer

## Summary

Redeemer fee is not configurable once deployed.

## Vulnerability Detail

The current implementation of Redeemer intended to configure the redemption fee using the function `setFee`.
In the function `setFee`, it checks the `feeChange` and apply the new fee only after the time set by `feeChange`.
But there is no function to set `feeChange` and it is always zero, so `setFee` will always revert.

I guess the writer intended to apply a fee change scheduling logic of the Lender contract but missed implementation of `scheduleFeeChange` and `blockFeeChange`.

```solidity
function setFee(uint256 f) external authorized(admin) returns (bool) {
    uint256 feeTime = feeChange; //@audit this is always zero and not changeable
    if (feeTime == 0) {
        revert Exception(23, 0, 0, address(0), address(0)); //@audit will always revert here
    } else if (feeTime < block.timestamp) {
        revert Exception(
            24,
            block.timestamp,
            feeTime,
            address(0),
            address(0)
        );
    } else if (f < MIN_FEENOMINATOR) {
        revert Exception(25, 0, 0, address(0), address(0));
    }
    feenominator = f;
    delete feeChange;
    emit SetFee(f);
    return true;
}
```

## Impact

Nobody can change the redemption fee once deployed.

## Code Snippet

https://github.com/sherlock-audit/2022-10-illuminate/blob/main/src/Redeemer.sol#L168
https://github.com/sherlock-audit/2022-10-illuminate/blob/main/src/Lender.sol#L813
https://github.com/sherlock-audit/2022-10-illuminate/blob/main/src/Lender.sol#L823

## Tool used

Manual Review

## Recommendation

Implement the `scheduleFeeChange` and `blockFeeChange` as in the Lender contract.

Duplicate of #34
